app/src/main/java/org/schabi/newpipe/util/StreamDialogEntry.java

```diff
@@ -203,4 +209,42 @@ private static void openChannelFragment(final Fragment fragment,
                 fragment.requireActivity().getSupportFragmentManager(),
                 item.getServiceId(), uploaderUrl, item.getUploaderName());
     }
+
+    /////////////////////////////////////////////
+    // helper functions                        //
+    /////////////////////////////////////////////
+
+    private interface InfoCallback {
+        void onInfo(SinglePlayQueue item);
+    }
+
+    private static void fetchItemInfoIfSparse(final Fragment fragment,
+            final StreamInfoItem item,
+            final InfoCallback callback) {
+        if (item.getDuration() < 0) {
```

Comment:
The duration would be `< 0` even for live stream, but in that case fetching the info before enqueueing would not be any useful since the duration would still be `< 0`. So I'd do it like this:
```suggestion
        if ((item.getStreamType() == StreamType.LIVE_STREAM || item.getStreamType() == StreamType.AUDIO_LIVE_STREAM) && item.getDuration() < 0) {
```